public/pages/HistoricalDetectorDetail/containers/HistoricalDetectorDetail/HistoricalDetectorDetail.tsx

```diff
@@ -0,0 +1,410 @@
+  waitForMs,
+  getRunningHistoricalDetectorCount,
+} from '../../utils/helpers';
+import { CoreStart } from '../../../../../../../src/core/public';
+import { CoreServicesContext } from '../../../../components/CoreServices/CoreServices';
+import { getClusterStats } from '../../../../redux/reducers/elasticsearch';
+import { GET_ALL_DETECTORS_QUERY_PARAMS } from '../../../utils/constants';
+
+export interface HistoricalDetectorRouterProps {
+  detectorId?: string;
+}
+interface HistoricalDetectorDetailProps
+  extends RouteComponentProps<HistoricalDetectorRouterProps> {}
+
+interface HistoricalDetectorDetailModalState {
+  isOpen: boolean;
+  action: HISTORICAL_DETECTOR_ACTION | undefined;
+}
+
+export const HistoricalDetectorDetail = (
+  props: HistoricalDetectorDetailProps
+) => {
+  const core = React.useContext(CoreServicesContext) as CoreStart;
+  const isDark = darkModeEnabled();
+  useHideSideNavBar(true, false);
+  const dispatch = useDispatch();
+  const detectorId: string = get(props, 'match.params.detectorId', '');
+
+  const adState = useSelector((state: AppState) => state.ad);
+  const allDetectors = adState.detectors;
+  const errorGettingDetectors = adState.errorMessage;
+  const detector = allDetectors[detectorId];
+  const monitors = useSelector((state: AppState) => state.alerting.monitors);
+  const monitor = get(monitors, `${detectorId}.0`);
+
+  // TODO: look further into whether or not we need to check for missing data or not
+  const isDetectorMissingData = false;
+
+  const [isLoadingDetector, setIsLoadingDetector] = useState<boolean>(true);
+  const [isStoppingDetector, setIsStoppingDetector] = useState<boolean>(false);
+  const [
+    historicalDetectorDetailModalState,
+    setHistoricalDetectorDetailModalState,
+  ] = useState<HistoricalDetectorDetailModalState>({
+    isOpen: false,
+    action: undefined,
+  });
+  const callout = getCallout(detector, isStoppingDetector);
+
+  useEffect(() => {
+    if (errorGettingDetectors) {
+      core.notifications.toasts.addDanger(
+        'Unable to get the historical detector'
+      );
+      props.history.push('/historical-detectors');
+    }
+  }, [errorGettingDetectors]);
+
+  useEffect(() => {
+    if (detector) {
+      core.chrome.setBreadcrumbs([
+        BREADCRUMBS.ANOMALY_DETECTOR,
+        BREADCRUMBS.HISTORICAL_DETECTORS,
+        { text: detector ? detector.name : '' },
+      ]);
+    }
+  }, [detector]);
+
+  const fetchDetector = async () => {
+    try {
+      await dispatch(getDetector(detectorId));
+      setIsLoadingDetector(false);
+    } catch {
+      core.notifications.toasts.addDanger(
+        'Error retrieving the historical detector'
+      );
+    }
+  };
+
+  // Try to get the detector initially
+  useEffect(() => {
+    if (detectorId) {
+      setIsLoadingDetector(true);
+      fetchDetector();
+    }
+  }, []);
+
+  // If detector is initialiazing or running: keep fetching every second to quickly update state/results/percentage bar, etc.
```

Comment:
Seems this description can't match the code, `HISTORICAL_DETECTOR_RESULT_REFRESH_RATE ` is 10 seconds

Reply:
Thanks, fixed.